File: store/src/main/java/com/nytimes/android/external/store3/base/impl/StoreBuilder.java
package com.nytimes.android.external.store3.base.impl;

import javax.annotation.Nonnull;


/**
 * Builder where there parser is used.
 */
public final class StoreBuilder {
    private StoreBuilder() {
    }

    @Nonnull
    @Deprecated
    public static <Raw> RealStoreBuilder<Raw, Raw, BarCode> builder() {
        return new RealStoreBuilder<>();
    }

    @Nonnull
    public static <Parsed> RealStoreBuilder<Parsed, Parsed, BarCode> barcode() {
        return new RealStoreBuilder<>();
    }

    @Nonnull
    public static <Key, Parsed> RealStoreBuilder<Parsed, Parsed, Key> key() {
        return new RealStoreBuilder<>();
    }

    @Nonnull
    public static <Key, Raw, Parsed> RealStoreBuilder<Raw, Parsed, Key> parsedWithKey() {
        return new RealStoreBuilder<>();
    }
}

File: store/src/main/java/com/nytimes/android/external/store3/base/impl/StoreBuilder.kt
package com.nytimes.android.external.store3.base.impl


/**
 * Builder where there parser is used.
 */
object StoreBuilder {

    @Deprecated("")
    fun <Raw> builder(): RealStoreBuilder<Raw, Raw, BarCode> {
        return RealStoreBuilder()
    }

    fun <Parsed> barcode(): RealStoreBuilder<Parsed, Parsed, BarCode> {
        return RealStoreBuilder()
    }

    fun <Key, Parsed> key(): RealStoreBuilder<Parsed, Parsed, Key> {
        return RealStoreBuilder()
    }

    fun <Key, Raw, Parsed> parsedWithKey(): RealStoreBuilder<Raw, Parsed, Key> {
        return RealStoreBuilder()
    }
}
